samples/nrf9160/http_update/modem_delta_update/src/main.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <stdio.h>
+#include <string.h>
+#include <bsd.h>
+#include <modem/bsdlib.h>
+#include <modem/at_cmd.h>
+#include "update.h"
+
+#define FOTA_TEST "FOTA-TEST"
+
+static char version[256];
+
+static bool is_test_firmware(void)
+{
+	static bool version_read;
+	int err;
+	
+	if (!version_read) {
+		err = at_cmd_write("AT+CGMR", version, sizeof(version),
+				NULL);
+		if (err != 0) {
+			printk("Unable to read modem version: %d\n", err);
+			return false;
+		}
+
+		if (strstr(version, CONFIG_SUPPORTED_BASE_VERSION) == NULL) {
+			printk("Unsupported base modem version: %s\n", version);
+			printk("Supported base version (set in prj.conf): %s\n",
+					CONFIG_SUPPORTED_BASE_VERSION);
+			return false;
+		}
+
+		version_read = true;
+	}
+
+	return (strstr(version, FOTA_TEST) != NULL);
```

Comment:
Minor: I don't think you need parentheses around the logic statement here. Resolve and ignore if you don't care.